app/models/course.rb

```diff
@@ -144,11 +144,11 @@ def new_draft_attributes
           class_name: "CourseEnrichment"
 
   scope :within, ->(range, origin:) do
-    joins(:sites).merge(Site.within(range, origin: origin))
+    joins(site_statuses: :site).merge(SiteStatus.where(status: %i[new_status running])).merge(Site.within(range, origin: origin))
   end
 
   scope :by_distance, ->(origin:) do
-    joins(:sites).merge(Site.by_distance(origin: origin))
+    joins(site_statuses: :site).merge(SiteStatus.where(status: %i[new_status running])).merge(Site.by_distance(origin: origin))
```

Comment:
`Course#sites` is a has many through association but it has a custom join condition on the `status`. I've pulled the condition out into the scope so that we get the same effect but AR doesn't need to set up a completely separate join to connect `sites`.

Reply:
Does `new_status`  have any affects?

I'm reading it as 
![image](https://user-images.githubusercontent.com/470137/135237431-d7e4240f-a46c-4a3d-8b95-f03d1c8b91fd.png)

Reply:
Thanks again @defong . That condition with `new_status` was moved from the join condition on the `Course#sites` association. So it appeared in the original query as:

```
INNER JOIN "site" ON "site"."id" = "site_statuses_course_join"."site_id" AND "course_site"."status" IN ('N', 'R')
```

As it's an inner join I don't _think_ the `course_site.status IN ('N', 'R')` changes the semantics of the original query. Merging the two joins does prevent you getting matches on courses that match the variation conditions across two different sites but I think that's the only change to the results here.

Does that make sense?

Reply:
Since it is the nested query `returns a list of ids`
Then it should be

 ```sql
course_site.status = 'R'
```
as `course_site` is incrementally costly from previously capped at 37 to now unlimited
the outter query disregards the unneeded resultsets


Whereas previously the nested query unneeded discards the resultsets
```sql
    # NOTE: Wide open join 
    INNER JOIN "course_site" ON "course_site"."course_id" = "course"."id"
    # NOTE: restricted join 
    INNER JOIN "site" ON "site"."id" = "site_statuses_course_join"."site_id" AND "course_site"."status" IN ('N', 'R')
    WHERE "provider"."recruitment_cycle_id" = 3
    # NOTE: removes `N`
    AND "course_site"."status" = 'R'
```

**In summary**
previously returns `returns a list of ids` without `N`
current returns `returns a list of ids` with `N` which can introduce a different problem

Reply:
@defong that makes sense, I've updated that condition to no longer include `new_status`.